### PR TITLE
Authentication via GitHub User

### PR DESCRIPTION
**Pull request by RodrigoPatrao**
Only authentication, no extra features.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -12,5 +12,6 @@
 //
 //= require jquery
 //= require jquery_ujs
+//= require bootstrap
```
Comment (filabreu) [inline app/assets/javascripts/application.js]: This is not really related to this feature and should be committed into other PR. If we ever need to revert this to use another frontend library, we would be obligated to also revert the OAuth feature. But that's ok for now, we can go on like this. Just a heads up for good practice in code committing.

File: app/assets/javascripts/home.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (filabreu) [inline app/assets/javascripts/home.coffee]: Don't commit empty files.

File: app/assets/javascripts/session.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (filabreu) [inline app/assets/javascripts/session.coffee]: Don't commit empty files.

File: app/assets/stylesheets/application.css.scss [hunk 4]
```diff
@@ -13,3 +13,9 @@
  *= require_tree .
  *= require_self
  */
+@import "bootstrap-sprockets";
+@import "bootstrap";
+@import "bootstrap/theme";
+body{
+  margin-top: 60px;
+}
```
Comment (filabreu) [inline app/assets/stylesheets/application.css.scss]: As the above commits about front-end library being not related to this feature PR, this should not be committed also. And we should not have any css in the application.css, this should be moved to other css file and imported in the application.css. Ideally our application.css should only have imports.

File: app/assets/stylesheets/home.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the home controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (filabreu) [inline app/assets/stylesheets/home.scss]: Don't commit empty files.

File: app/assets/stylesheets/session.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the session controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (filabreu) [inline app/assets/stylesheets/session.scss]: Don't commit empty files.

File: app/helpers/home_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
+end
```
Comment (filabreu) [inline app/helpers/home_helper.rb]: Don't commit empty files.

File: app/helpers/session_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module SessionHelper
+end
```
Comment (filabreu) [inline app/helpers/session_helper.rb]: Don't commit empty files.

File: config/database.yml [hunk 9]
```diff
@@ -63,7 +63,7 @@ test:
   database: decide_test
   username: postgres
   password:
-  host: db
+  host: localhost
```
Comment (filabreu) [inline config/database.yml:67]: Don't commit changes to database.yml. This can break development environment for other users. This is a must fix!

File: app/controllers/session_controller.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def destroy
+    if current_user
+      session.delete(:user_id)
+      flash[:success] = 'Até logo!'
+    end
+    redirect_to root_path
+  end
+end
```
Comment (filabreu) [inline app/controllers/session_controller.rb:18]: We should have a path (or even a controller) for each of our providers. As we are only using Github, we can keep it this way. We should also store several providers uuid, so perhaps we should have another model to store user OAuth data. We can work more on that later.

**Review by filabreu — CHANGES_REQUESTED** [review]
Most of my comments were regarding commits context. We should always commit code contextualized with the current feature, we should not mix things up. But we can merge the PR with those, no problem. Just a heads up for following commits.

But there are a lot of unneeded empty files committed. Those should be removed. And changes to database.yml shouldn't be commited, as we can break things to other developers.

File: app/controllers/session_controller.rb (hunk 10 above)
Comment (RodrigoPatrao) [reply app/controllers/session_controller.rb:18]: It was made this way, I've focused only on GitHub authentication. Should I include another providers now, or we'll handle this later?

Comment (filabreu) [reply app/controllers/session_controller.rb:18]: We can handle this later. No problem! :+1:
